kubernetes/controller/internal/configuration/config.go

```diff
@@ -0,0 +1,123 @@
+// Package configuration provides functionality to load and manage OCM configurations
+// from Kubernetes resources (Secrets and ConfigMaps).
+package configuration
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"sort"
+	"strings"
+
+	corev1 "k8s.io/api/core/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	genericv1 "ocm.software/open-component-model/bindings/go/configuration/generic/v1/spec"
+	"ocm.software/open-component-model/bindings/go/runtime"
+	"ocm.software/open-component-model/kubernetes/controller/api/v1alpha1"
+)
+
+// GetConfigFromSecret extracts and decodes OCM configuration from a Kubernetes Secret.
+// It looks for configuration data under the OCMConfigKey.
+func GetConfigFromSecret(secret *corev1.Secret) (*genericv1.Config, error) {
+	data, ok := secret.Data[v1alpha1.OCMConfigKey]
```

Comment:
i think if the key is not found we should error so people know they configured the wrong secret